dhall/src/Dhall/Parser/Expression.hs

```diff
@@ -323,13 +323,18 @@ parsers embedded = Parsers {..}
             f <-    (Some <$ _Some <* nonemptyWhitespace)
                 <|> return id
             a <- noted importExpression_
-            b <- Text.Megaparsec.many (try (nonemptyWhitespace *> noted importExpression_))
-            return (foldl app (f a) b)
+            bs <- Text.Megaparsec.many . try $ do
+                (sep, _) <- Text.Megaparsec.match nonemptyWhitespace
+                b <- importExpression_
+                return (sep, b)
+            return (foldl app (f a) bs)
```

Comment:
It's not due to this patch, but why don't we use `foldl'` here?

Reply:
Because Haskell should be strict by default :laughing: 

Might as well change that while I'm at it.